### PR TITLE
VACMS-11874 Remove flipper logic for PDF modal in Find Forms

### PR DESCRIPTION
**Pull request by randimays**
## Summary
The `find_forms_show_pdf_modal` feature flipper was previously used to show a download modal for a form if the feature flipper was on. In the search results screenshot below, when `Download VA Form 10-10M` was clicked and the feature flipper is on, the modal from the second screenshot would be shown. Clicking on the `Download VA Form 10-10M` link on the modal would lead to the prod page for the form shown in the third screenshot.

The fallback behavior (if the flipper was off) would be to show the prod page for the form when clicking `Download VA Form 10-10M` on the search result.

### Result in find forms search results
Screenshot 2024-02-20 at 12 46 33 PM

### Modal to show
Screenshot 2024-02-20 at 12 46 37 PM

### Prod page for form
Screenshot 2024-02-20 at 12 46 49 PM

## Desired behavior
We want to continue the behavior of first showing the modal before the Download button is clicked, and remove the fallback behavior.

## Related issue(s)
- https://github.com/department-of-veterans-affairs/va.gov-cms/issues/11867
- https://github.com/department-of-veterans-affairs/va.gov-cms/issues/11874

## Testing done
Tested manually on desktop and mobile.

## Screenshots
See screenshots in the description.

## Acceptance criteria
- [x] https://github.com/department-of-veterans-affairs/va.gov-cms/issues/11874 is complete, and modal shows when clicking to download a form from search results on https://www.va.gov/find-forms/ with find_forms_show_pdf_modal, with flipper disabled
 All references to the flipper in vets-website, if any, are removed (should happen via 11874)

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] Linting warnings have been addressed
- [x] Screenshot of the developed feature is added
- [x] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### CODE REVIEW

**Review by chriskim2311 — DISMISSED** [review]
LGTM!

File: src/applications/find-forms/tests/widgets/DownloadPDFGuidance.unit.spec.js [hunk 1]
```diff
@@ -3,162 +3,106 @@ import ReactDOM from 'react-dom';
...
-    let domStub;
-    let reduxStore;
+  before(() => {
+    domStub = sinon.stub(ReactDOM, 'render').returns(<div />);
+  });
 
-    before(() => {
-      domStub = sinon.stub(ReactDOM, 'render').returns(<div />);
+  after(() => {
+    domStub.restore();
```
Comment (micahchiang) [inline src/applications/find-forms/tests/widgets/DownloadPDFGuidance.unit.spec.js:18]: Thank you :+1:

**Review by micahchiang — DISMISSED** [review]

**Comment by randimays** [comment]
@chriskim2311 @micahchiang Did you have more feedback, or is this ready to merge?

**Review by chriskim2311 — DISMISSED** [review]
Re-Approved!

**Review by micahchiang — APPROVED** [review]

**Pull request merged**